### PR TITLE
Add memory info to get system memory usage

### PR DESCRIPTION
**Pull request by yixinglu**
refer to vesoft-inc/nebula-graph#1052

### CODE REVIEW

File: src/common/base/Memory.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#include <cstdint>
+
+#include "common/cpp/helpers.h"
+
+namespace nebula {
+
+class MemInfo final : protected cpp::NonCopyable, protected cpp::NonMovable {
+public:
+    MemInfo() noexcept {
+        sysinfo(&info_);
```
Comment (critical27) [inline src/common/base/Memory.h:21]: Could consider to check the return value here.

File: src/common/base/Memory.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    uint64_t freeInKB() const {
+        return (info_.freeram * info_.mem_unit) >> 10;
+    }
+
+    uint64_t usedInKB() const {
+        return totalInKB() - freeInKB();
+    }
+
+    bool hitsHighWatermark(float ratio = 0.9f) const {
```
Comment (critical27) [inline src/common/base/Memory.h:36]: Better set the same default value with `system_memory_high_watermark_ratio`.

File: src/common/base/Memory.h (hunk 1 above)
Comment (yixinglu) [reply src/common/base/Memory.h]: now Log error when return value is -1.

File: src/common/base/Memory.h (hunk 2 above)
Comment (yixinglu) [reply src/common/base/Memory.h]: reset to 0.8f

**Review by bright-starry-sky — APPROVED** [review]
LGTM

File: src/common/base/Memory.cpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    NG_RETURN_IF_ERROR(mem->init());
+    return mem;
+}
+
+MemInfo::MemInfo() noexcept {
+    info_ = std::make_unique<struct sysinfo>();
+}
+
+Status MemInfo::init() {
+    if (sysinfo(info_.get()) == -1) {
```
Comment (bright-starry-sky) [inline src/common/base/Memory.cpp:26]: What's about `if (sysinfo(info_.get()) != 0)` ?

**Review by dutor — APPROVED** [review]
LGTM.

**Pull request merged**